crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,11 +286,129 @@ impl Executor {
                     }
                 }
             }
+        };
+
+        match &reason {
+            Ok(reason) => {
+                info!(reason, "shutting down");
+            }
+            Err(reason) => {
+                error!(%reason, "shutting down");
+            }
+        };
+
+        let mut total_bundles_drained_successfully = 0;
+        let shutdown_logic = async {
```

Comment:
Please report how many bundles need to be drained before you enter this shutdown logic